Update from auto-update/renku-ui-2.14.0 (#2865)
SwissDataScienceCenter/renku: CHANGELOG.rst

Renku ``0.21.0`` brings the ability to handle expiring Gitlab tokens in user sessions and various bug fixes. 

User-Facing Changes
~~~~~~~~~~~~~~~~~~~

**✨ Improvements**

> * 🎨 **UI**: Improve layout on different pages: project creation, starting sessions, file browser.

Reviewer: ```suggestion
* 🎨 **UI**: Improve the layout of the project creation, session start, and file browser pages.
```

**🐞 Bug Fixes**

> * 📃 **UI**: Update broken links local projects, to documentation, and add more links to useful resources.

Reviewer 2: `Update broken links to local projects and documentation and add more links to useful resources.`

?

> * 📃 **UI**: Update broken links to local projects and documentation, and add more links to useful resources.

Reviewer: Where were these broken links, and where are the new links to useful resources?

Reviewer 3: Scattered in many places -- workflows, notebooks, other project pages... Here is the full list https://github.com/SwissDataScienceCenter/renku-ui/pull/2199/files

I tried to condense a few minor fixes here. Since they are minor, I didn't want to write 3 separate sentences. But here are the actual changes:
1. Update broken links (result of documentation changes) https://github.com/SwissDataScienceCenter/renku-ui/pull/2199/files
2. Add link to what is an S3 storage object https://github.com/SwissDataScienceCenter/renku-ui/pull/2207/files
3. Fix link to local projects https://github.com/SwissDataScienceCenter/renku-ui/pull/2209/files

* 🚀 **UI**: Fix glitches with autosave and improve layout of session pages.
* 🔦 **UI**: Prevent flashing inputs when forking a project.

Internal Changes
~~~~~~~~~~~~~~~~

**Improvements**

* **Gateway**: Add endpoint for refreshing expiring Gitlab tokens
* **Sessions**: Check LFS size and available disk space before cloning

> * **UI**: Receive notifications when sessions change

Reviewer 2: `Added support for websocket-based notifications when session state changes`

Reviewer 3: I would like to keep the changelog as non-technical as possible, leaving out the WebSocket detail.

Reviewer: I agree with Sean. This is already in the "non-user facing" part of the changelog, where the goal of the changelog is primarily so we (the development team) can keep track of and refer to what was released when. We should make the descriptions specific enough so we remember what the point is referring to.
